model/pod/pod.go

```diff
@@ -87,3 +121,44 @@ func (p *Pod) Remove() error {
 		},
 	)
 }
+
+// UpdateStatus updates the pod status.
+func (p *Pod) UpdateStatus(s Status) error {
```

Comment:
would you consider checking for invalid state changes? For example, changing from Initializing to Provisioned could be invalid.

Reply:
should we check the status is different from the current status?
For instance, we wouldn't want to update the terminated time if it's already been terminated.

Reply:
Eh, I don't think it's worth guarding against unlikely programmatic errors like state transitions. I don't think that it's worth coding all the possible correct state transitions in this function (e.g. in the host lifecycle, you can transition from several different phases to "running" or "terminated" or "decommissioned"). It's also unlikely that the state diagram would do something nonsensical (e.g. from running to initializing) unless there's a serious flaw in logic.

I think it's not worth adding more complexity to deal with the possibility that the status is identical. That would be a programmatic error since you would typically check the status before you set it anyways.